// @aws-sdk/client-athena/dist-es/models/enums.js
export const S3AclOption = {
    BUCKET_OWNER_FULL_CONTROL: "BUCKET_OWNER_FULL_CONTROL",
};
export const AuthenticationType = {
    DIRECTORY_IDENTITY: "DIRECTORY_IDENTITY",
};
export const EncryptionOption = {
    CSE_KMS: "CSE_KMS",
    SSE_KMS: "SSE_KMS",
    SSE_S3: "SSE_S3",
};
export const StatementType = {
    DDL: "DDL",
    DML: "DML",
    UTILITY: "UTILITY",
};
export const QueryExecutionState = {
    CANCELLED: "CANCELLED",
    FAILED: "FAILED",
    QUEUED: "QUEUED",
    RUNNING: "RUNNING",
    SUCCEEDED: "SUCCEEDED",
};
export const DataCatalogType = {
    FEDERATED: "FEDERATED",
    GLUE: "GLUE",
    HIVE: "HIVE",
    LAMBDA: "LAMBDA",
};
export const ConnectionType = {
    BIGQUERY: "BIGQUERY",
    CMDB: "CMDB",
    DATALAKEGEN2: "DATALAKEGEN2",
    DB2: "DB2",
    DB2AS400: "DB2AS400",
    DOCUMENTDB: "DOCUMENTDB",
    DYNAMODB: "DYNAMODB",
    GOOGLECLOUDSTORAGE: "GOOGLECLOUDSTORAGE",
    HBASE: "HBASE",
    MYSQL: "MYSQL",
    OPENSEARCH: "OPENSEARCH",
    ORACLE: "ORACLE",
    POSTGRESQL: "POSTGRESQL",
    REDSHIFT: "REDSHIFT",
    SAPHANA: "SAPHANA",
    SNOWFLAKE: "SNOWFLAKE",
    SQLSERVER: "SQLSERVER",
    SYNAPSE: "SYNAPSE",
    TIMESTREAM: "TIMESTREAM",
    TPCDS: "TPCDS",
};
export const DataCatalogStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_FAILED_CLEANUP_COMPLETE: "CREATE_FAILED_CLEANUP_COMPLETE",
    CREATE_FAILED_CLEANUP_FAILED: "CREATE_FAILED_CLEANUP_FAILED",
    CREATE_FAILED_CLEANUP_IN_PROGRESS: "CREATE_FAILED_CLEANUP_IN_PROGRESS",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_COMPLETE: "DELETE_COMPLETE",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
};
export const ThrottleReason = {
    CONCURRENT_QUERY_LIMIT_EXCEEDED: "CONCURRENT_QUERY_LIMIT_EXCEEDED",
};
export const NotebookType = {
    IPYNB: "IPYNB",
};
export const CalculationExecutionState = {
    CANCELED: "CANCELED",
    CANCELING: "CANCELING",
    COMPLETED: "COMPLETED",
    CREATED: "CREATED",
    CREATING: "CREATING",
    FAILED: "FAILED",
    QUEUED: "QUEUED",
    RUNNING: "RUNNING",
};
export const CapacityAllocationStatus = {
    FAILED: "FAILED",
    PENDING: "PENDING",
    SUCCEEDED: "SUCCEEDED",
};
export const CapacityReservationStatus = {
    ACTIVE: "ACTIVE",
    CANCELLED: "CANCELLED",
    CANCELLING: "CANCELLING",
    FAILED: "FAILED",
    PENDING: "PENDING",
    UPDATE_PENDING: "UPDATE_PENDING",
};
export const QueryResultType = {
    DATA_MANIFEST: "DATA_MANIFEST",
    DATA_ROWS: "DATA_ROWS",
};
export const ColumnNullable = {
    NOT_NULL: "NOT_NULL",
    NULLABLE: "NULLABLE",
    UNKNOWN: "UNKNOWN",
};
export const SessionState = {
    BUSY: "BUSY",
    CREATED: "CREATED",
    CREATING: "CREATING",
    DEGRADED: "DEGRADED",
    FAILED: "FAILED",
    IDLE: "IDLE",
    TERMINATED: "TERMINATED",
    TERMINATING: "TERMINATING",
};
export const WorkGroupState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const ExecutorState = {
    CREATED: "CREATED",
    CREATING: "CREATING",
    FAILED: "FAILED",
    REGISTERED: "REGISTERED",
    TERMINATED: "TERMINATED",
    TERMINATING: "TERMINATING",
};
export const ExecutorType = {
    COORDINATOR: "COORDINATOR",
    GATEWAY: "GATEWAY",
    WORKER: "WORKER",
};
