// @aws-sdk/client-athena/dist-es/models/errors.js
import { AthenaServiceException as __BaseException } from "./AthenaServiceException";
export class InternalServerException extends __BaseException {
    name = "InternalServerException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidRequestException extends __BaseException {
    name = "InvalidRequestException";
    $fault = "client";
    AthenaErrorCode;
    Message;
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
        this.AthenaErrorCode = opts.AthenaErrorCode;
        this.Message = opts.Message;
    }
}
export class TooManyRequestsException extends __BaseException {
    name = "TooManyRequestsException";
    $fault = "client";
    Message;
    Reason;
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.Message = opts.Message;
        this.Reason = opts.Reason;
    }
}
export class ResourceNotFoundException extends __BaseException {
    name = "ResourceNotFoundException";
    $fault = "client";
    Message;
    ResourceName;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
        this.ResourceName = opts.ResourceName;
    }
}
export class MetadataException extends __BaseException {
    name = "MetadataException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "MetadataException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MetadataException.prototype);
        this.Message = opts.Message;
    }
}
export class SessionAlreadyExistsException extends __BaseException {
    name = "SessionAlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "SessionAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SessionAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
